src/inc/sift-events/sift-events.php

```diff
@@ -565,12 +569,17 @@ public static function update_or_create_order( string $order_id, \WC_Order $orde
 		}
 
 		// Determine user and session context.
-		$user_id   = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
+		$user_id   = get_current_user_id() ?? wp_get_current_user()->ID ?? null; // Check first for logged-in user.
 		$is_system = ! $create_order && str_starts_with( sanitize_title( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ?? '' ) ), 'WordPress' ); // Check if this is an order update via system action.
+		$is_admin  = 1 === $user_id;
 
 		// Figure out if it should use the session ID if no logged-in user exists.
+		if ( ! $user_id || $is_admin ) {
+			$user_id = $is_system || $is_admin ? $order->get_user_id() : null; // Use order user ID only for system actions.
+		}
+
 		if ( ! $user_id ) {
-			$user_id = $is_system ? $order->get_user_id() : null; // Use order user ID only for system actions.
+			$user_id = \WC()->session->get_customer_unique_id();
```

Comment:
See my comment about it. This is equivalent to use `get_current_user_id()`, except it could also return a session_id, which is not what we want. I would not use that,...

Reply:
Removed.